feat: support Go comments within HTML tag attributes

jackielii opened the pull request:
## Summary
Implements support for single-line (`//`) and multi-line (`/* */`) Go comments within HTML element attributes, addressing issue #866.

## Problem
Previously, templ did not allow Go comments within HTML tag attributes, requiring developers to place comments above the element which wasn't always convenient for documenting specific attributes.

## Solution
This PR adds support for Go-style comments directly within HTML tag attributes. The comments are parsed and stored in the AST but excluded from the generated output.

### Example
```templ
<form
  class="flex"
  // This comment explains why the following attribute is necessary
  action="javascript:void(0)"
  /* Multi-line comment
     for more detailed explanations */
  method="post"
>
```

## Changes
- Added `AttributeComment` type to represent comments within attributes
- Updated parser to recognize and handle comments between attributes  
- Updated generator to skip comments during code generation
- Added comprehensive test coverage in `generator/test-attribute-comments/`

## Testing
- Added test case following the existing test pattern in the generator folder
- Test verifies that comments are properly excluded from generated output
- All existing tests continue to pass

Fixes #866

jackielii committed:
feat: add support for Go comments within HTML tag attributes

Implements support for single-line (//) and multi-line (/* */) Go comments
within HTML element attributes, addressing issue #866.

Changes:
- Add AttributeComment type to represent comments within attributes
- Update parser to recognize and handle comments between attributes
- Update generator to skip comments during code generation
- Add comprehensive test coverage

This allows developers to add explanatory comments directly within HTML tags
for better code documentation without affecting the generated output.

Fixes #866

jackielii committed:
fix: formatter handling of attribute comments

jackielii committed:
comment according to a-h standard.

jackielii committed:
merge

jackielii committed:
Merge branch 'main' into feat/attribute-comments

jackielii committed:
add range and edge cases tests

jackielii committed:
fix test: somehow it needs weird format, probably because <link> shouldnt appear here

jackielii commented:
@a-h anything else you need?

a-h commented:
@joerdav - we discussed this one last Friday, right?

IIRC, we said we'd merge it.

I was a bit reluctant because it's hard to walk something back once you've added it, and it changes the syntax highlighting and some other stuff.

jackielii commented:
I've been using this patch over a year now. I'm exclusive Neovim user, the treesitter already supports this. I think it's a natural extension to existing syntax: support commenting inside the HTML tag is less unexpected than not supporting it.

a-h commented:
Agreed. I'm merging this today, just fixing the conflicts.

a-h committed:
merge: resolve conflict with main branch

Keep both the attribute comment tests from this branch and the unquoted
attribute tests from main. Update comment test expectations to include
ValueRange and Range fields added in main.

a-h committed:
style: format elementparser_test.go

a-h committed:
chore: generate
